Lib/sqlite3/test/regression.py

```diff
@@ -383,72 +385,30 @@ def CheckDelIsolation_levelSegfault(self):
         with self.assertRaises(AttributeError):
             del self.con.isolation_level
 
+    def CheckBpo37347(self):
+        class Printer:
+            def log(self, *args):
+                return sqlite.SQLITE_OK
 
-class UnhashableFunc:
-    __hash__ = None
+        for method in [self.con.set_trace_callback,
+                       functools.partial(self.con.set_progress_handler, n=1),
+                       self.con.set_authorizer]:
+            printer_instance = Printer()
+            method(printer_instance.log)
+            log = printer_instance.log
+            refcnt_before = sys.getrefcount(log)
+            method(log)
+            refcnt_after = sys.getrefcount(log)
```

Comment:
We should try to avoid writing tests that use `sys.getrefcount`. Does this test fail if you pass `-R ::` to `regrtest.py`?

Reply:
Nope. But it can be rewritten to fall to segmentation fault.

Reply:
Sounds good to me. Not crashing is the behavior we want to test anyway.

Reply:
Then it is done in d1b57ff.